chapters/02-filesystem.Rmd

```diff
@@ -1,16 +1,137 @@
+  * if you are looking at RStudio with no project open, you'll see the working directory listed at the top of the Console as `~/` (or Windows equivalent)
+
+* text (image): create a new project
+  * File -> New Project -> New Directory -> New Project
+  * Directory name: `sassafras`
+  * Create project as subdirectory of: `~/Desktop` (or the equivalent on a Windows machine?); may need to use Browse button
+  * Click "Create Project"
+
+* text: what happens after creating a project?
+  * screen refreshes and path to working directory changes (ends in name of project you just created)
+* code: `pwd` and `ls -a` in Terminal to confirm working directory and show files in project
+* text: interpreting output:
+  * if creating a new project, creates a folder in the specified location that includes hidden files containing information about your project
+  * RStudio references that information to make it easier for us to focus on writing code (includes working directory and our preferences)
+
+* text: why aren't we creating a project to use in later chapters?
+  * while creating your project, you may have noticed the option to create a package, as these are more specialized, structured types of projects
+  * we won't be working with RStudio projects in subsequent chapters, but they demonstrate a few important principles that will help you understand how and why we perform certain actions later
+
+## Managing files and directories with `fs`
+
+* text: introduce {fs} as the recommended approach for working with the filesystem (files, directories, and paths)
+  * could use shell, but sometimes commands for common tasks are complex, and it's inconvenient to move back and forth between your R Console and shell
+  * could use same functions in base R includes, 
+  * {fs} combines the best features of both shell and base R filesystem management, creating a standardized interface for interacting with your files and directories with intuitive names for the functions involved
+* text: {fs} provides additional advantages when creating R packages:
+  * vectorization
+  * predictable return values
+  * explicit failure
+  * doesn't rely on native system encoding (makes everything UTF-8)
+
+* text: we'll look at some of the most common functions from {fs} relevant to development of a data analysis package:
+  * identifying and locating data files
+  * saving intermediate and final output files
+* text: {fs} functions have three different categories, prefaced with:
+  * `dir_` operate on directories
+  * `file_ ` operate on files
+  * `path_` operate on paths
+  * while covering all useful functions is outside the scope of thsi chapter, we'll demonstrate a few to help you understand their breadth of applications
+
+* code: `dir_ls()` to list files in current directory
+  * as with Unix commands, providing no argument defaults to the current working directory
+* code: `dir_ls("Desktop")`: indicate a different directory by including the path as an argument 
```

Comment:
This will fail if they're inside the `sassafras` project already - maybe that's fine, to demonstrate that a relative path is relative to your current place. Or could just use `dir_ls("~/Desktop")`.